packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -0,0 +1,319 @@
+                    return (caps as Capabilities.Capabilities)['browserstack.accessibility']
+                }
+            } else if (capType === 'deviceName') {
+                if ((caps as Capabilities.Capabilities)['bstack:options'] && (caps as Capabilities.Capabilities)['bstack:options']?.deviceName) {
+                    return (caps as Capabilities.Capabilities)['bstack:options']?.deviceName
+                } else if ((caps as Capabilities.Capabilities)['bstack:options'] && (caps as Capabilities.Capabilities)['bstack:options']?.device) {
+                    return (caps as Capabilities.Capabilities)['bstack:options']?.device
+                } else if ((caps as Capabilities.Capabilities)['appium:deviceName']) {
+                    return (caps as Capabilities.Capabilities)['appium:deviceName']
+                }
+            } else if (capType === 'goog:chromeOptions' && (caps as Capabilities.Capabilities)['goog:chromeOptions']) {
+                return (caps as Capabilities.Capabilities)['goog:chromeOptions']
+            } else {
+                const bstackOptions = (caps as Capabilities.Capabilities)['bstack:options']
+                if ( bstackOptions && bstackOptions?.[capType as keyof Capabilities.BrowserStackCapabilities]) {
+                    return bstackOptions?.[capType as keyof Capabilities.BrowserStackCapabilities]
+                } else if ((caps as Capabilities.Capabilities)[legacyCapType as keyof Capabilities.Capabilities]) {
+                    return (caps as Capabilities.Capabilities)[legacyCapType as keyof Capabilities.Capabilities]
+                }
+            }
+        }
+    }
+
+    async before () {
+        this._accessibility = isTrue(this._getCapabilityValue(this._caps, 'accessibility', 'browserstack.accessibility'))
+
+        if (isBrowserstackSession(this._browser) && isAccessibilityAutomationSession(this._accessibility)) {
+            const deviceName = this._getCapabilityValue(this._caps, 'deviceName', 'device')
+            const chromeOptions = this._getCapabilityValue(this._caps, 'goog:chromeOptions', '')
+
+            this._accessibility = validateCapsWithA11y(deviceName, this._platformA11yMeta, chromeOptions)
+        }
+
+        (this._browser as BrowserStack).getAccessibilityResultsSummary = async () => {
+            return await getA11yResultsSummary((this._browser as WebdriverIO.Browser), isBrowserstackSession(this._browser), this._accessibility)
+        }
+
+        (this._browser as BrowserStack).getAccessibilityResults = async () => {
+            return await getA11yResults((this._browser as WebdriverIO.Browser), isBrowserstackSession(this._browser), this._accessibility)
+        }
+    }
+
+    async beforeTest (suiteTitle: string | undefined, test: Frameworks.Test) {
+        if (this._framework !== 'mocha' || !this._browser) {
+            return
+        }
+
+        if (!this.shouldRunTestHooks(this._browser, this._accessibility)) {
+            return
+        }
```

Comment:
Let's just merge this:

```suggestion
        if (
            this._framework !== 'mocha' ||
            !this._browser ||
            !this.shouldRunTestHooks(this._browser, this._accessibility)
        ) {
            return
        }
```